File: src/lib/Logger.php
<?php
/**
 * Simple Logger for Application
 */

class Logger {
    private static $levels = [
        'DEBUG' => 0,
        'INFO' => 1,
        'WARNING' => 2,
        'ERROR' => 3
    ];

    /**
     * Log message to file
     */
    private static function log($level, $message, $context = []) {
        if (self::$levels[$level] < self::$levels[LOG_LEVEL]) {
            return;
        }

        $timestamp = date('Y-m-d H:i:s');
        $logFile = LOG_PATH . '/' . date('Y-m-d') . '.log';

        $contextStr = !empty($context) ? ' ' . json_encode($context) : '';
        $logMessage = sprintf(
            "[%s] [%s] %s%s\n",
            $timestamp,
            $level,
            $message,
            $contextStr
        );

        file_put_contents($logFile, $logMessage, FILE_APPEND);
    }

    public static function debug($message, $context = []) {
        self::log('DEBUG', $message, $context);
    }

    public static function info($message, $context = []) {
        self::log('INFO', $message, $context);
    }

    public static function warning($message, $context = []) {
        self::log('WARNING', $message, $context);
    }

    public static function error($message, $context = []) {
        self::log('ERROR', $message, $context);
    }
}

File: moodle-dmn-detection/backend/utils/logger.py
"""
Logging Configuration
Sets up logging for the application
"""

import logging
import sys
from logging.handlers import RotatingFileHandler
import os


def setup_logger(name: str, level: str = 'INFO') -> logging.Logger:
    """
    Setup logger with console and file handlers

    Args:
        name: Logger name
        level: Logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)

    Returns:
        Configured logger
    """
    logger = logging.getLogger(name)

    # Convert level string to logging level
    log_level = getattr(logging, level.upper(), logging.INFO)
    logger.setLevel(log_level)

    # Avoid duplicate handlers
    if logger.handlers:
        return logger

    # Console handler
    console_handler = logging.StreamHandler(sys.stdout)
    console_handler.setLevel(log_level)

    # File handler (rotating)
    log_dir = os.path.join(
        os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
        'logs'
    )
    os.makedirs(log_dir, exist_ok=True)

    log_file = os.path.join(log_dir, f'{name}.log')
    file_handler = RotatingFileHandler(
        log_file,
        maxBytes=10 * 1024 * 1024,  # 10MB
        backupCount=5
    )
    file_handler.setLevel(log_level)

    # Formatter
    formatter = logging.Formatter(
        '%(asctime)s - %(name)s - %(levelname)s - %(message)s',
        datefmt='%Y-%m-%d %H:%M:%S'
    )

    console_handler.setFormatter(formatter)
    file_handler.setFormatter(formatter)

    logger.addHandler(console_handler)
    logger.addHandler(file_handler)

    return logger
